src/org/omegat/gui/main/MainWindow.java

```diff
@@ -106,8 +107,10 @@
  */
 @SuppressWarnings("serial")
 public class MainWindow extends JFrame implements IMainWindow {
-    public final BaseMainWindowMenu menu;
+    public static final String menuClassID = "OmegaTMainWindowMenu";
+    public static final String toolbarClassID = "OmegaTMainWindowToolbar";
 
+    public BaseMainWindowMenu menu;
```

Comment:
Unless it becomes possible to change it on the fly without rebooting the application  I would prefer to keep this variable final (especially if you keep it public!). Yes I know that it requires some changes, see below

Reply:
Thank you for the suggestion. It is public because `EditorController` class acces menu field to enable/disable menu items.

I think there is a space we can define a better menu API for components/modules, to add/remove menu item, enable/disable specified menu item, etc.

Reply:
Yes but this is not a reason to allow something like ` ((MainWindow) Core.getMainWindow()).menu = new Menu.... `

And yes, you are true, an API would be better to give an abstraction about what is allowed or not to do with the menu.

This conducts me to a point which I did not know where to put, as it is linked to new UI but not to this patch.
Actually both MainWindowMenu and MainWindowNewMenu have a method which quotes the sub-menus one by one. That means that old and new UI _may_ contain a distinct list of menus if the code differs: is it an intentional behaviour? I think you should find a way to put this code in BaseMainWindowMenu, or put the list in BaseMainWindowMenu to ensure it is the same in all sub-implementations. Unless I am wrong and it was an intentional behaviour? Do you have ideas about any menu which should appear in old UI and not in new one or the contrary?

Reply:
> I think you should find a way to put this code in BaseMainWindowMenu, or put the list in BaseMainWindowMenu to ensure it is the same in all sub-implementations. Unless I am wrong and it was an intentional behaviour?

I'd like to take a way __puting the list in BaseMainWindowMenu__